app/src/main/java/org/mozilla/tv/firefox/MainActivity.kt

```diff
@@ -116,7 +118,7 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener, Media
 
         // We can't do anything if the intent does not contain valid data, so short
         val intentData = IntentValidator.validate(this, unsafeIntent.toSafeIntent()) ?: return
-        serviceLocator.screenController.showBrowserScreenForUrl(supportFragmentManager, intentData.url)
+        serviceLocator.intentLiveData.value = intentData
```

Comment:
Can we leave a comment here explaining why we throw this in a `LiveData` instead of handling it immediately?